### PR TITLE
Final integration test changes

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Final changes needed for #16525

See note in review.

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml [hunk 1]
```diff
@@ -92,6 +96,18 @@ t4_LookupByKeyMissingAuthorization = scenario do
     lookupByKey @TheContract4 ([alice],100)
     abort "t4 finished"
 
+Why does this fail? It doesn't fail when converted to scripts, as from my perspective - shouldn't.
+bob is explicitly an observer of the contract with that key, why should he not be able to look it up?
+-}
```
Comment (samuel-williams-da) [inline compiler/damlc/tests/daml-test-files/AuthFailure.daml:101]: Unsure whats happening with this test, would love some clarity

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml [hunk 2]
```diff
@@ -104,11 +120,10 @@ template TheContract5
...
-  bob <- getParty "Bob-t5"
-  cid <- submit alice $ do
-    create (TheContract5 with s = alice)
-  submit alice $ do
-    exercise cid TheChoice5 with cs = [bob]
-    abort "t5 finished"
+t5_ExerciseMissingAuthorization : Script ()
+t5_ExerciseMissingAuthorization = script do
+  alice <- allocateParty "Alice-t5"
+  bob <- allocateParty "Bob-t5"
```
Comment (samuel-williams-da) [inline compiler/damlc/tests/daml-test-files/AuthFailure.daml:126]: Drop the numbers, use source loc

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml (hunk 1 above)
Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/AuthFailure.daml:101]: Why should it fail?

Comment (remyhaemmerle-da) [reply compiler/damlc/tests/daml-test-files/AuthFailure.daml:101]: It is important in the test you do not change fetch / lookup by a query which work very differently.

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml [hunk 3]
```diff
@@ -57,18 +60,16 @@ template TheContractBuilder3
...
       do create TheContract3 with s = s2
 
--- @ERROR failed since none of the stakeholders 'Alice-t3' is in the authorizing set 'Bob-t3'
-t3_FetchMissingAuthorization = scenario do
-  alice <- getParty "Alice-t3"
-  bob <- getParty "Bob-t3"
-  builder <- submit alice $ do
-    create (TheContractBuilder3 with s = alice; c = bob)
+t3_QueryMissingAuthorization : Script ()
+t3_QueryMissingAuthorization = script do
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/AuthFailure.daml:64]: Here we should test fetch not query.

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks. 
There are two test to fix.

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml [hunk 4]
```diff
@@ -13,13 +15,13 @@ template TheContract1
...
-  submit mach $ do
-    create (TheContract1 with s = ivy)
-    abort "t1 finished"
+-- @ERROR range=20:1-20:30; failed due to a missing authorization from 'Ivy'
+t1_CreateMissingAuthorization : Script ()
+t1_CreateMissingAuthorization = script do
+  mach <- allocateParty "Mach"
+  ivy <- allocateParty "Ivy"
+  mach `submit` createCmd (TheContract1 with s = ivy)
+  abort "t1 finished"
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/AuthFailure.daml:24]: (very minor remark) 

Originally the `abort` is always done inside the ledger (inside the Update monad), here you make it outside. 
In fact I prefer it like that. For the sake of consistency I will make it consistently outstide.

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml [hunk 5]
```diff
@@ -57,18 +59,29 @@ template TheContractBuilder3
...
+  with
+    bob : Party
+    cid : ContractId TheContract3
+  where
+    signatory bob
+    choice Call3 : ()
+      controller bob
+      do
+        _ <- fetch @TheContract3 cid
+        abort "t3 finished"
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/AuthFailure.daml:72]:
```suggestion
        fetch @TheContract3 cid
```

File: compiler/damlc/tests/daml-test-files/AuthFailure.daml [hunk 6]
```diff
@@ -57,18 +59,29 @@ template TheContractBuilder3
...
+    createCmd (TheContractBuilder3 with s = alice; c = bob)
   -- note: we use `builder` so `cid` is visible to bob, although he is not a stakeholder
-  cid <- submit bob $ do
-    exercise builder TheChoice with s2 = alice
-  submit bob $ do
-    _ <- fetch @TheContract3 cid
-    abort "t3 finished"
+  cid <- bob `submit`
+    exerciseCmd builder TheChoice with s2 = alice
+  bob `submit` createAndExerciseCmd (Test3 bob cid) Call3
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/AuthFailure.daml:84]:
```suggestion
  _ <- bob `submit` createAndExerciseCmd (Test3 bob cid) Call3
  abort "t3 finished"
```

And similarly for the following cases.

File: compiler/damlc/tests/daml-test-files/Loop.daml [hunk 7]
```diff
@@ -20,30 +18,16 @@ template Helper
...
-ledgerMachineFromScenarioSubmit = scenario do
-  alice <- getParty "p"
-  alice `submit` createAndExercise (Helper alice) Loop
-
--- @ERROR range=33:1-33:40; Evaluation timed out after 3 seconds
-ledgerMachineFromScenarioSubmitMustFail = scenario do
-  alice <- getParty "p"
-  alice `submitMustFail` createAndExercise (Helper alice) Loop
-
--- @ERROR range=38:1-38:14; Evaluation timed out after 3 seconds
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/Loop.daml:37]: :fire:

**Review by remyhaemmerle-da — APPROVED** [review]

File: compiler/damlc/tests/daml-test-files/ScenarioError.daml [hunk 8]
```diff
@@ -1,5 +1,6 @@
 -- This tests that defining a scenario triggers an error when scenarios are disabled.
 -- @ERROR Scenarios are no longer supported.
+-- Remove when scenarios are fully deleted (TODO - make/find a ticket)
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/daml-test-files/ScenarioError.daml:3]:
```suggestion
-- TODO https://github.com/digital-asset/daml/issues/16711
--   Remove when scenarios are fully deleted
```

File: compiler/damlc/tests/daml-test-files/SemanticsEvalOrderSince1_11.daml [hunk 9]
```diff
@@ -1,41 +0,0 @@
-
--- | Test operational semantics for 1.11 features.
---
--- @SINCE-LF 1.11
-module SemanticsEvalOrderSince1_11 where
```
[the remark this answers is not in the document]
Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/SemanticsEvalOrderSince1_11.daml:5]: Honestly I think we made this decision in an lf meeting like 3 weeks ago, i don't recall the exact justification
I'll take a look

[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply compiler/damlc/tests/daml-test-files/SemanticsEvalOrderSince1_11.daml:5]: :+1:

File: compiler/damlc/tests/daml-test-files/ScenarioError.daml (hunk 8 above)
Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/ScenarioError.daml:3]: Is this the right ticket? I dont think this relates to the canton work

Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/ScenarioError.daml:3]: Going to link to [16583](https://github.com/digital-asset/daml/issues/16583)

File: compiler/damlc/tests/daml-test-files/SemanticsEvalOrderSince1_11.daml (hunk 9 above)
[the remark this answers is not in the document]
Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/SemanticsEvalOrderSince1_11.daml:5]: Had a look, this exact test is already done in scala, and much better. Thats why we removed it.

File: compiler/damlc/tests/daml-test-files/ScenarioError.daml (hunk 8 above)
Comment (remyhaemmerle-da) [reply compiler/damlc/tests/daml-test-files/ScenarioError.daml]: My bad, 16583 is the good one.

**Pull request merged**